File: coin_flip_streaks/java/CoinFlipStreaks.java
import java.util.Random;

public class CoinFlipStreaks {

    public static void main(String[] args) {
        int numOfStreaks = 0;
        Random random = new Random();

        for (int i = 0; i < 10000; i++) {

            char[] list = new char[100];
            for (int j = 0; j < 100; j++) {
                if (random.nextInt(2) == 0) {
                    list[j] = 'H';
                } else {
                    list[j] = 'T';
                }
            }

            char prev = list[0];
            int count = 1;
            for (int curr = 1; curr < list.length; curr++) {
                if (list[curr] == prev) {
                    count++;
                } else {
                    prev = list[curr];
                    count = 1;
                }

                if (count == 6) {
                    numOfStreaks++;
                    count = 0;
                }
            }
        }

        System.out.printf("Chance of streak: %s%%", numOfStreaks / 100);
    }
}

File: coin_flip_streaks/python/coin_flip_streaks.py
import random

numberOfStreaks = 0

for experimentNumber in range(10000):
    # Code that creates a list of 100 'heads' or 'tails' values.
    result = []
    for i in range(100):
        if random.randint(0, 1) == 0:
            result.append('H')
        else:
            result.append('T')
            
    # Code that checks if there is a streak of 6 heads or tails in a row.
    count = 1
    begin = result[0]
    for cur in result[1:]:
        if cur == begin:
            count += 1
        else:
            begin = cur
            count = 1
        
        if count == 6:
            numberOfStreaks += 1
            count = 0

print('Chance of streak: %s%%' % (numberOfStreaks / 100))
